src/runtime/arch_tinygoriscv.go

```diff
@@ -36,7 +36,8 @@ func procUnpin() {
 
 func waitForEvents() {
 	mask := riscv.DisableInterrupts()
-	if !runqueue.Empty() {
+	runqueue := schedulerRunQueue()
+	if !(runqueue != nil && runqueue.Empty()) {
```

Comment:
Nit: `runqueue == nil || !runqueue.Empty()` is clearer to me, but I understand why you write it like this.

Nittier: `if rq := schedulerRunQueue(); rq == nil || !rq.Empty() {`